presto-main/src/main/java/com/facebook/presto/server/ServerInfoResource.java

```diff
@@ -108,6 +112,14 @@ public NodeState getServerState()
             return SHUTTING_DOWN;
         }
         else {
+            if (coordinator) {
```

Comment:
Should we guard this change with a config flag? This change isn't backwards compatible and if any load balancer is relying on this endpoint on the coordinator (as well as the `query-manager.required-workers`), it might have to change its behavior.

Reply:
As discussed offline, it's not really supported on coordinator right now.